### DISCUSSION TITLE
How to establish a reasonable training set when the system charged

### DISCUSSION THREAD

**Discussion by liubaoshuai1402** [ACEsuit/mace#693] [Q&A] [answered]
Dear developers,

I am trying to fine-tuning Foundation model, MACE-mp-0b.
But I am not sure whether I am on the right way to collect my training set.
Here are my questions:
(1) **more configurations, more better model?** Since the name is called as fine-tuning, I am worried that too many configurations will make the model worse. If this true, how many is enough?
(2) **whether wrong configurations should be collected?**  my research is about insulator in the form of crystal but at high temperatures. So I wonder whether the amorphous structures at higher temperatures should be collected in MACE just like other ML-software do, for example deepMD-kit.

**Comment by gabor1** [comment]
(1) I think it is generally true the more configurations you have that are relevant to what you want to study, the better your model will be. How many you need, is hard to say without knowing more precisely what you are trying to achieve. When you fine-tune, often a few dozen configurations are enough to get a meaningful improvement in accuracy for specific systems. 

(2) yes, I think higher energy structures like the ones you describe will be informative. But the best structures are those that you directly care about, so if you want to study crystals at high temperatures, do simulations with the foundation model of that system, collect configurations, and fine tune on those. One thing you might run into is that the foundation model might cause your structure to melt at a temperature that is below the real melting point. This should be improved with just a few dozen configs collected at high temperature (both before and after the melting), so if you then continue to collect configurations with your find-tuned model, you should be able to simulate your high temperature solid without melting it at too low a temperature.

**Reply by liubaoshuai1402** [reply]
Dear professor, thanks for your kindly reply.
I have other questions now, I am sorry that I don't ask them at one time.
**Could MACE handle a charged system? More specially, a system containing one proton or more?**
The reason I ask the question is because that I find the free_energy of a charged system (I get from OUTCAR after VASP calculation) is naturally lower than a no-charged system though they have the same atoms and similar configurations. In my systems, the value is about 10 eV. I get charged systems by control the total value electron numbers.(for VASP, using the NELECT in INCAR)
In fact, I did some defect formation-energy(FE) research.
<img width="525" alt="ec" src="https://github.com/user-attachments/assets/4f6a5a2b-0790-4ad1-92d1-a92b1bc311b9">
The FE of proton is only comparable to a neutral hydrogen after considering the Fermi energy (I often take it as VBM+a variable from 0)
I aim to study the proton behavior in the insulator.
**Therefore, (1) Is it OK that I simply mix the no-charged configurations (no H atoms, can also be called perfect bulk) and the charged configurations(containing proton) to form my training set without a free energy correction?**
I am not meaning the electrostatic interactions corrections. 
I think the free_energy of configurations containing one proton should be add a value of VBM (value band max, we assume the Fermi level = 0).
In this way, the free_energy should be comparable to neutral one. Of course, if electrostatic interactions corrections are also considered, it will be better.
I also find MACE in lammps only support atom-type, atomic.
So, It seems no possible to specify a H atoms as proton or neutral atom as I please.
Therefore, I think if I want H behaves as proton in my systems, I have to consider the charged configurations.
Another possible resolution I can think, all the configurations is add a proton when performs DFT calculation. Then  No correction is needed.
(2) **If i study proton, E0s of H should be calculated in a charged cell containg only a H atom?**  
@gabor1 
Hoping for your instruction!

**Reply by gabor1** [reply, marked as answer]
the current version of mace does not do _explicit_ electrostatic long range interactions. But this doesn't mean that you can't study protons in insulators, especially ones which have very good screening (e.g. water), so that the long range contribution is small. if your DFT data contains protons, mace will learn them. in practice I think it does help if your overall cell is neutral. we do quite well on proton hopping in water for example. you don't need to indicate in lammps that your H is a proton, its behaviour will be inferred from its local environment. 

there are plans to treat charges, fermi levels, long range electrostatics more properly, stay tuned.
